esmf_regrid/schemes.py

```diff
@@ -170,10 +172,19 @@ def _regrid_rectilinear_to_rectilinear__prepare(src_grid_cube, tgt_grid_cube):
 def _regrid_rectilinear_to_rectilinear__perform(src_cube, regrid_info, mdtol):
     grid_x_dim, grid_y_dim, grid_x, grid_y, regridder = regrid_info
 
-    # Perform regridding with realised data for the moment. This may be changed
-    # in future to handle src_cube.lazy_data.
-    new_data = _regrid_along_grid_dims(
-        regridder, src_cube.data, grid_x_dim, grid_y_dim, mdtol
+    regrid = functools.partial(
+        _regrid_along_grid_dims,
+        regridder,
+        grid_x_dim=grid_x_dim,
+        grid_y_dim=grid_y_dim,
+        mdtol=mdtol,
+    )
+
+    new_data = map_complete_blocks(
+        src_cube,
+        regrid,
+        (grid_x_dim, grid_y_dim),
```

Comment:
A bit of documentation would be useful here - a comment to explain why you are passing the horizontal dims as non-chunkable to the map function.

Reply:
I've added a comment explaining.